way-cooler-bg/src/main.rs

```diff
@@ -0,0 +1,285 @@
+#[macro_use]
+extern crate wayland_client;
+
+extern crate tempfile;
+
+extern crate byteorder;
+extern crate image;
+
+use std::env;
+use std::process::exit;
+use image::open;
+use std::mem::transmute;
+use std::os::unix::io::AsRawFd;
+use std::io::Write;
+
+use wayland_client::wayland::get_display;
+use wayland_client::wayland::compositor::{WlCompositor, WlSurface};
+use wayland_client::wayland::shell::WlShell;
+use wayland_client::wayland::shm::{WlBuffer, WlShm, WlShmFormat};
+use wayland_client::wayland::seat::{WlSeat, WlPointerEvent};
+use wayland_client::cursor::load_theme;
+use wayland_client::{EventIterator, Proxy};
+
+use byteorder::{NativeEndian, WriteBytesExt};
+
+wayland_env!(WaylandEnv,
+             compositor: WlCompositor,
+             shell: WlShell,
+             shm: WlShm,
+             seat: WlSeat
+);
+
+type BufferResult = Result<WlBuffer, ()>;
+
+/// Hack to deal with edge case between the two cursor buffer types...
+enum CursorBuffer {
+    /// The buffer has been leaked into memory.
+    /// This should eventually be fixed, but since it's just the cursor
+    /// it's not a big issue.
+    Null,
+    /// The buffer for the cursor, this can be destroyed so it should last
+    /// as long as the program (or until you replace it with another).
+    Buf(WlBuffer)
+}
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+/// Holds the bytes to represent a colored background.
+/// To be written into a wayland surface.
+pub struct Color(pub [u8; 4]);
+
+impl Color {
+    /// Generate a new color out of a u32.
+    /// E.G: 0xFFFFFF
+    pub fn from_u32(color: u32) -> Self {
+        unsafe { Color(transmute(color)) }
+    }
+
+    pub fn as_u32(&self) -> u32 {
+        unsafe { transmute(self.0)}
+    }
+}
+
+fn main() {
+    let args: Vec<_> = env::args().collect();
+    if args.len() < 3 {
+        println!("Please supply either a file path or a color (written in hex)");
+        println!("Please also supply a path to a cursor image");
```

Comment:
We _could_, but then we run into the same issue as before where the background program will fail if it can't find an appropriate cursor to use from the default theme. A better way to think of this cursor is a "falback" option if a cursor can't be found (like on Ubuntu). If you want I can change it so you don't have to specify a fallback option, but since we _should_ already have one (check the `assets/` folder) I decided to just keep it required. 

Also, this program should _never_ be called directly by the user, it should be called by way cooler / lua, so ultimately this decision will not affect the user thankfully.